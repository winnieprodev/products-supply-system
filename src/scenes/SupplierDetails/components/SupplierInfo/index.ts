export { default as SupplierInfo } from "./SupplierInfo";
