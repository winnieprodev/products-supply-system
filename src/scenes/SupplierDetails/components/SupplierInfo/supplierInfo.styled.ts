import styled from "styled-components";
import { monoTextColor, secondaryColorDarker } from "styles/colors";

export const Wrapper = styled.div`
  border-radius: 11px;
  align-items: center;
  background: rgba(196, 196, 196, 0.1);
  padding: 10px 20px;
  margin-bottom: 20px;
  position: relative;
`;

export const Section = styled.div`
  display: flex;
  padding: 2px 5px;
  align-items: center;
`;

export const MonoSection = styled.div`
  display: flex;
  padding: 2px;
  align-items: center;
  text-align: center;
  width: 70px;
  height: 55px;
  background: white;
  border-radius: 10px;
  box-shadow: 0 0 5px ${secondaryColorDarker};
`;

export const MonoText = styled.span`
  margin: auto;
  color: ${monoTextColor};
  font-size: 35px;
  font-weight: bold;
  text-transform: uppercase;
`;

export const Title = styled.span`
  padding-left: 15px;
  padding-right: 15px;
  color: #0b0b0b;
  font-size: 24px;
  font-weight: 700;
`;

export const ContactSection = styled.div`
  padding-left: 15px;
  padding-right: 15px;
  display: flex;
  align-items: center;
`;

export const ContactGroup = styled.div`
  display: flex;
  align-items: center;
  margin-top: 5px;
  margin-right: 10px;
`;

export const PhoneText = styled.span`
  padding-left: 5px;
  padding-right: 5px;
  color: #000000;
  font-size: 14px;
  font-weight: 400;
`;

export const EmailText = styled.span`
  padding-left: 5px;
  padding-right: 5px;
  color: #000000;
  font-size: 14px;
  font-weight: 400;
  text-decoration: underline;
`;

export const PhoneIcon = styled.div`
  font-size: 15px;
  margin-right: 2px;
  color: #000000;
  opacity: 0.5;
  transform: scaleX(-1);
`;

export const Icon = styled.div`
  font-size: 15px;
  margin-right: 2px;
  color: #000000;
  opacity: 0.5;
`;

export const SpecialGroup = styled.div`
  display: flex;
  align-items: center;
  margin-top: 20px;
  margin-right: 10px;
`;

export const SpecialText = styled.span`
  padding-left: 5px;
  padding-right: 5px;
  color: #484848;
  font-size: 10px;
`;
