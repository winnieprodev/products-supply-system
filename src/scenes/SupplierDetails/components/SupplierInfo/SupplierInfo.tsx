import { EnvelopeIcon, PhoneIcon, WarningIcon } from "components";
import { Supplier } from "models";
import React from "react";

import * as Styled from "./supplierInfo.styled";

interface OwnProps {
  info: Supplier;
}

const SupplierInfo: React.FC<React.PropsWithChildren<OwnProps>> = ({
  info,
}) => (
  <Styled.Wrapper>
    <Styled.Section>
      <Styled.MonoSection>
        <Styled.MonoText>{info.mono}</Styled.MonoText>
      </Styled.MonoSection>
      <div>
        <Styled.Title>{info.title}</Styled.Title>
        <Styled.ContactSection>
          <Styled.ContactGroup>
            <Styled.PhoneIcon>
              <PhoneIcon />
            </Styled.PhoneIcon>
            <Styled.PhoneText>{info.phone}</Styled.PhoneText>
          </Styled.ContactGroup>
          <Styled.ContactGroup>
            <Styled.Icon>
              <EnvelopeIcon />
            </Styled.Icon>
            <Styled.EmailText>{info.email}</Styled.EmailText>
          </Styled.ContactGroup>
        </Styled.ContactSection>
      </div>
    </Styled.Section>
    {info.remarks && (
      <Styled.SpecialGroup>
        <Styled.Icon>
          <WarningIcon />
        </Styled.Icon>
        <Styled.SpecialText>
          Special remarks for <b>{info.title}</b>: {info.remarks}
        </Styled.SpecialText>
      </Styled.SpecialGroup>
    )}
  </Styled.Wrapper>
);

export default React.memo(SupplierInfo);
