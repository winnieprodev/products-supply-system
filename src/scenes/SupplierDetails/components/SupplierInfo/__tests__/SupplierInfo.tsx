import { render } from "@testing-library/react";

import SupplierInfo from "../SupplierInfo";

describe("SupplierInfo", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <SupplierInfo
          info={{
            id: "",
            mono: "BF",
            title: "Ben's Food",
            phone: "67772869",
            email: "fsorder@benfoods.com",
          }}
        />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
