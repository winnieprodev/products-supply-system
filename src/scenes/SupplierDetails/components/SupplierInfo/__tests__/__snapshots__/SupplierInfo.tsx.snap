// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierInfo Snapshot tests it matches the snapshot 1`] = `
.c0 {
  border-radius: 11px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(196,196,196,0.1);
  padding: 10px 20px;
  margin-bottom: 20px;
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 2px 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 2px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  width: 70px;
  height: 55px;
  background: white;
  border-radius: 10px;
  box-shadow: 0 0 5px #E7E7E7;
}

.c3 {
  margin: auto;
  color: rgba(0,0,0,0.25);
  font-size: 35px;
  font-weight: bold;
  text-transform: uppercase;
}

.c4 {
  padding-left: 15px;
  padding-right: 15px;
  color: #0b0b0b;
  font-size: 24px;
  font-weight: 700;
}

.c5 {
  padding-left: 15px;
  padding-right: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 5px;
  margin-right: 10px;
}

.c8 {
  padding-left: 5px;
  padding-right: 5px;
  color: #000000;
  font-size: 14px;
  font-weight: 400;
}

.c10 {
  padding-left: 5px;
  padding-right: 5px;
  color: #000000;
  font-size: 14px;
  font-weight: 400;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7 {
  font-size: 15px;
  margin-right: 2px;
  color: #000000;
  opacity: 0.5;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
}

.c9 {
  font-size: 15px;
  margin-right: 2px;
  color: #000000;
  opacity: 0.5;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        BF
      </span>
    </div>
    <div>
      <span
        class="c4"
      >
        Ben's Food
      </span>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-phone fa-w-16 "
              data-icon="phone"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                fill="currentColor"
              />
            </svg>
          </div>
          <span
            class="c8"
          >
            67772869
          </span>
        </div>
        <div
          class="c6"
        >
          <div
            class="c9"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope fa-w-16 "
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                fill="currentColor"
              />
            </svg>
          </div>
          <span
            class="c10"
          >
            fsorder@benfoods.com
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
